### ISSUE TITLE
Error on Oni load

### ISSUE THREAD

**Issue by CrossR** [onivim/oni#1798]
Upon loading Oni, using the latest Master, I get this error:

```
Uncaught (in promise) TypeError: Cannot read property 'request' of undefined
    at NeovimInstance.eval (NeovimInstance.ts?e8a2:538)
    at NeovimAutoCommands.eval (NeovimAutoCommands.ts?f73a:116)
    at Generator.next (<anonymous>)
    at eval (563:8)
    at Promise (<anonymous>)
    at __awaiter (563:4)
    at NeovimAutoCommands.executeAutoCommand (NeovimAutoCommands.ts?f73a:115)
    at NeovimEditor.enter (NeovimEditor.tsx?b7dc:669)
    at OniEditor.enter (OniEditor.tsx?ed1d:189)
    at AugmentedWindow.enter (WindowManager.ts?db84:95)
eval @ NeovimInstance.ts?e8a2:538
(anonymous) @ NeovimAutoCommands.ts?f73a:116
(anonymous) @ 563:8
__awaiter @ 563:4
executeAutoCommand @ NeovimAutoCommands.ts?f73a:115
enter @ NeovimEditor.tsx?b7dc:669
enter @ OniEditor.tsx?ed1d:189
enter @ WindowManager.ts?db84:95
_focusNewSplit @ WindowManager.ts?db84:287
createSplit @ WindowManager.ts?db84:204
(anonymous) @ startEditors.ts?d033:56
(anonymous) @ 301:9
__awaiter @ 301:5
startEditors @ startEditors.ts?d033:25
(anonymous) @ index.tsx?0c97:185
(anonymous) @ 200:16
__awaiter @ 200:12
initializeAllEditors @ index.tsx?0c97:184
(anonymous) @ index.tsx?0c97:205
fulfilled @ 200:13
```

Following back the stack trace, I think its due to the changes that were added in #1786.
Specifically, the following line:
https://github.com/onivim/oni/blob/6d9cd90e1c20bc2c4ae3ce087f8695d5856457b0/browser/src/Editor/NeovimEditor/NeovimEditor.tsx#L669

Which I think is calling on Neovim, when it isn't fully setup yet.
If I pause on exception, the error occurs when calling `exists('#FocusGained')`, so I am assuming that is the cause of the error.

**Comment by bryphe** [comment]
@CrossR , nice find! I saw this came up but didn't realize it was a regression - I think you're right that #1786 breaks this. I'll take a look at it today. Thanks for catching this!

**Comment by Akin909** [comment]
Seems the error persists despite #1800, with the addition of `notifyGoal` of undefined

<img width="460" alt="screen shot 2018-03-15 at 10 32 40" src="https://user-images.githubusercontent.com/22454918/37458297-8c04006c-283c-11e8-98ae-2f94b857b4c1.png">

<img width="826" alt="screen shot 2018-03-15 at 10 29 42" src="https://user-images.githubusercontent.com/22454918/37458300-91c9407a-283c-11e8-8ad9-3e2ba24191c9.png">

Sorry for adding another one in there, I just wanted to raise that the issue persists unfortunately. Regarding the `focusedGained` and lost functionality this feature exists for the workspace which is what the git status uses to update the branch I wonder if that functionality could be used to dispatch the focusedGained?

**Comment by bryphe** [comment]
Thanks @Akin909 for the update! I just fixed this in #1800 

Those errors should've been caught in our automation - so there was a hole in our _safety net_. That fix addresses the hole (we weren't always checking the logs for errors), so hopefully it'll be harder to introduce regressions like this in the future 👍

**Comment by bryphe** [comment]
(The fix just got merged in - so might be worth pulling the latest master if you're still hitting it)
